Tests/UZKArchiveTestCase.m

```diff
@@ -61,7 +61,8 @@ - (void)setUp {
                            @"Test File B.jpg",
                            @"Test File C.m4a",
                            @"NotAZip-PK-ContentsUnknown",
-                           @"Modified CRC Archive.zip"];
+                           @"Modified CRC Archive.zip",
+                           @"Test Permissions Archive.zip",];
```

Comment:
It might be better to generate the zip file instead of committing it to the repo. I've started doing this whenever I can so the repo size doesn't grow too much. It's pretty easy if you use one of the `-[UZKArchiveTestCase archiveWithFiles:]` overloads. I verified just now that the system `zip` shell command preserves permissions (on Catalina, at least, but I doubt it's changed). So you can write out some test text files and use `NSFileManager` to modify the permissions before archiving them. As with my suggestion above, I can help with this. I'd suggest writing a helper method in your test case class to write the files with the permissions you want, perhaps even naming the files with the octal permissions you're archiving (e.g. `777.txt`, `722.txt`).

Reply:
Next, I may be unable to participate in the revision due to work problems. I need your help :(